packages/effect/src/Number.ts

```diff
@@ -493,3 +493,57 @@ export const parse = (s: string): Option<number> => {
     ? option.none
     : option.some(n)
 }
+
+/**
+ * Takes a string and returns an `Option` of `number`.
+ *
+ * If the string is empty or invalid characters,
+ * it returns `Option.none`. Otherwise, it attempts to convert the string to a number
+ * and returns `Option.some(number)`.
+ *
+ * @param s - The string to be converted to a `number`.
+ *
+ * @example
+ * import { fromString } from "effect/Number"
+ *
+ * assert.deepStrictEqual(fromString("42"), option.some(42))
+ * assert.deepStrictEqual(fromString(""), option.none)
+ * assert.deepStrictEqual(fromString("not a number"), option.none)
+ *
+ * @category conversions
+ * @since 2.4.12
+ */
+export const fromString = (s: string): Option<number> => {
+  const n = Number(s)
+  return s.trim() === ""
+    ? option.none
+    : Number.isNaN(n)
+    ? option.none
+    : option.some(n)
+}
+
+/**
+ * Takes a `bigint` and returns an `Option` of `number`.
+ *
+ * If the `bigint` is outside the safe integer range for JavaScript (`Number.MAX_SAFE_INTEGER`
+ * and `Number.MIN_SAFE_INTEGER`), it returns `Option.none`. Otherwise, it converts the `bigint`
+ * to a number and returns `Option.some(number)`.
+ *
+ * @param b - The `bigint` to be converted to a `number`.
+ *
+ * @example
+ * import { fromBigInt } from "effect/Number"
+ *
+ * assert.deepStrictEqual(fromBigInt(BigInt(42)), option.some(42))
+ * assert.deepStrictEqual(fromBigInt(BigInt(Number.MAX_SAFE_INTEGER) + BigInt(1)), option.none)
+ * assert.deepStrictEqual(fromBigInt(BigInt(Number.MIN_SAFE_INTEGER) - BigInt(1)), option.none)
+ *
+ * @category conversions
+ * @since 2.4.12
+ */
+export const fromBigInt = (b: bigint): Option<number> => {
```

Comment:
I think we should move this to the BigInt module as `toNumber`.

Reply:
Thank you for your feedback.

When I checked `BigInt.toNumber`, that function was already existed. So I deleted to remove `Number.fromBigInt` and strengthened the test of `BigInt.toNumber`